crates/types/src/consensus.rs

```diff
@@ -248,16 +251,24 @@ impl<TYPES: NodeType> Consensus<TYPES> {
         Ok(())
     }
 
-    /// Update the last proposed view.
+    /// Update the last proposal.
     ///
     /// # Errors
     /// Can return an error when the new view_number is not higher than the existing proposed view number.
-    pub fn update_last_proposed_view(&mut self, view_number: TYPES::Time) -> Result<()> {
+    pub fn update_last_proposed_view(
+        &mut self,
+        proposal: Proposal<TYPES, QuorumProposal<TYPES>>,
+    ) -> Result<()> {
         ensure!(
-            view_number > self.last_proposed_view,
+            proposal.data.view_number()
+                > self
+                    .last_proposals
+                    .first_key_value()
```

Comment:
should this be `last_key_value` for the proposal with the highest key (view number)?